src/main/java/com/google/sps/travelbud/EventServlet.java

```diff
@@ -35,12 +35,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
 
     String endpoint = request.getPathInfo();
+    String eventName = request.getParameter("name");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     Gson gson = new Gson();
 
-    // return the list of events
-    if (endpoint == null || endpoint == "/") {
+    if (eventName != null) {
+      List<Event> filteredEvents = Event.filter(datastore, eventName);
+      response.getWriter().println(gson.toJson(filteredEvents));
+    } else if (endpoint == null) {
```

Comment:
Suggestion: don't overcomplicate
```suggestion
if (endpoint == null) {
  return Event.getAll(datastore).stream()
      .filter(e -> eventName == null || e.name.toLowerCase().contains(eventName.toLowerCase()))
      .collect(Collectors.toList());
```

You can delete the filtering function from the Event class.